liblangutil/ParserBase.cpp

```diff
@@ -85,6 +85,45 @@ void ParserBase::expectToken(Token _value, bool _advance)
 		m_scanner->next();
 }
 
+void ParserBase::expectTokenRecoveryDelete(Token _value, bool _advance)
+{
+	Token tok = m_scanner->currentToken();
+	if (tok != _value)
+	{
+		auto tokenName = [this](Token _token)
+		{
+			if (_token == Token::Identifier)
+				return string("identifier");
+			else if (_token == Token::EOS)
+				return string("end of source");
+			else if (TokenTraits::isReservedKeyword(_token))
+				return string("reserved keyword '") + TokenTraits::friendlyName(_token) + "'";
+			else if (TokenTraits::isElementaryTypeName(_token)) //for the sake of accuracy in reporting
+			{
+				ElementaryTypeNameToken elemTypeName = m_scanner->currentElementaryTypeNameToken();
+				return string("'") + elemTypeName.toString() + "'";
+			}
+			else
+				return string("'") + TokenTraits::friendlyName(_token) + "'";
+		};
+
+		Token _token;
+		SourceLocation const& location = SourceLocation{position(), endPosition(), source()};
+		for (_token = m_scanner->currentToken(); _token != _value && _token != Token::EOS; _token = m_scanner->next()) ;
+
+		if (_token == Token::EOS)
+			fatalParserError(location,
+							string("Expected ") + tokenName(_value) + string(" but got ") + tokenName(tok) +
+							"; can not find token to synchronize to.");
+		else
+			parserError(location,
+						string("Expected ") + tokenName(_value) + string(" but got ") + tokenName(tok) +
+						"; deleted tokens until offset " + string(std::to_string(position()) + string(".")));
```

Comment:
The offset (as in SourceLocation) is only used internally and the user usually always sees line/column informations. It should be the same here, too. I also think this information shouldn't be part of the actual parserError message but rather as an additional informational message *below* this parser error message.

Reply:
Good idea! I've now  added a parserWarning which indicates the range of text deleted. Please double check.

Reply:
it's actually purely a notice to the user how the compiler attempts to recover from the parsing error (which is already of type error, so there is no reason to issue a message of type warning. it should rather be notice/informational) -- IMHO :-)

Reply:
Is this message intended for testing while working on the pull request or do you think this is information the user should care about?

Reply:
@chriseth I think this information that both a user and an IDE may about. 

An IDE can show which parts were not parsed. 

@christianparpart I am not seeing any informational messages anywhere. Warning was the lowest that I saw. 

I was trying to work within the framework that is there, and even here, I had to add a parserWarning. However if the consensus is to create this as an Informational message and not track it like warnings and errors are tracked, that would be great.

Do we want to go this route?

Reply:
I'd probably have introduced a Notice/Info level; (and would have checked if we have other places that could make use of it, but that would then be an isolated PR, I'm thinking of "this is a beta-release compiler"-message for example). -- Let's see what @chriseth is saying :)

Reply:
I think this is an internal detail we should not bother the user with. An IDE might use that information, but I'm pretty sure nobody will ever implement that. If that information is requested at some point, we might add it again. At the moment and seeing how people use the compiler, this will only confuse them because they assume something is wrong with the code at the point where the warning is printed, while the actual error is somewhere else.

Reply:
> but I'm pretty sure nobody will ever implement that

Well my plan was not just do the solidity part but also the IDE integration part for say vscode-solidity and/or remix to make sure this is used. And of course if I were to implement it, I'd use it ;-)

That said, it's not worth fussing over. It does simplify things and moves things forward.